Moving weaver tests into their own classes

James-Frowen opened the pull request:
Splitting region from `WeaverTest` into separate classes

vis2k commented:
is this based on the other PR?
there seem to be two big weaver test PRs now.
I already asked in the other PR: is it a good idea that we don't have the hidden folder anymore?
if yes, why?
we had this for a reason before

James-Frowen commented:
This is the first PR, the other one (https://github.com/vis2k/Mirror/pull/1706) has one extra commit that needs to be reviewed. The goal of these 2 PR is to make it easier to add new weaver tests.

The scripts in the `WeaverTests~` folder are the ones that are hidden, and they should still be hidden in this PR. I have just moved the `WeaverTests~` folder.

In master all of the weaver tests were in a single file: `Assets/Mirror/Tests/Editor/WeaverTest.cs` which has regions to groups the tests. This PR splits those regions into their own classes, and uses a new `WeaverTests` class as a bass class so that they can all use the same TestFixtures.

I have also moved all the weaver tests into a sub folder 
Assets/Mirror/Tests/Editor/ → Assets/Mirror/Tests/Editor/Weaver/

paulpach commented:
Merged manually to avoid breaking other prs
